### DISCUSSION TITLE
Syncing albums with iCloud shared library

### DISCUSSION THREAD

**Discussion by RhetTbull** [RhetTbull/osxphotos#1304] [Show and tell]
iCloud shared library syncs keywords and other metadata between shared libraries but does not sync albums. With [v0.65.0](https://github.com/RhetTbull/osxphotos/releases/tag/v0.65.0) of osxphotos, I've added several new features include an `--album` option to `osxphotos batch-edit`. This allows you to now sync albums between shared libraries using the following workflow.

On the first user's machine (that has albums you want to sync), run:

```bash
 osxphotos batch-edit --verbose --keyword "{album?album:{folder_album}}"
```

which will write the album name in form `album:Folder/Album` to the keyword field.  Then on the other user's machine:

```bash
osxphotos batch-edit --verbose --album "{keyword|filter(startswith album:)|sslice(6:)}" --split-folder "/" 
```

reads the album name from the keyword field and splits it into folder and album name and adds the photo to the album, creating album and folders as necessary.

Both commands can be run on each user's machine to sync albums between shared libraries. The commands can also be combined into a single command that can periodically run on each user's machine:

```bash
osxphotos batch-edit --verbose --album "{keyword|filter(startswith album:)|sslice(6:)}" --split-folder "/" --keyword "{album?album:{folder_album}}"
```

Note that if a photo is removed from an album, this will not remove it (nor will it remove the `album:AlbumName` keyword that gets added) so that's one current limitation. This could be sovled with some more complex logic or a custom `osxphotos run` script but I've not tackled that yet.

**Comment by Audefleda** [comment]
@RhetTbull Thanks a ton, that was exactly what I needed!

Is there a way how I can run this command automatically? I created an intelligent folder with all the photos that are candidates for synching, but I did not find a way how to select this folder automatically via a script.

**Reply by RhetTbull** [reply]
@Audefleda Unfortunately `batch-edit` doesn't have a way to run against a specific album at the moment but it's something I plan to add. You have to manually select the photos. This is by design as I didn't want it to be possible to accidentally apply edits to the entire library. I do plan to allow selection of photos in a specific album. Also, OSXPhotos cannot read Smart Albums at the moment though I'm making progress on this.

**Reply by Audefleda** [reply]
Thanks a lot, for your quick reply. For the time being, I can easily live with that restriction, the synching of albums already saved me hours of tedious work...
